pkg/controller/daemon/daemon_controller.go

```diff
@@ -535,6 +535,15 @@ func (dsc *DaemonSetsController) updatePod(old, cur interface{}) {
 		return
 	}
 
+	if curPod.DeletionTimestamp != nil {
```

Comment:
Yes, it might reduce the nonsense syncloop.
The only risk I have considered is that once handler(updatePod or deletePod) returns before lowering of expectation, we will also lose the chance to satisfy the expectation.
@k82cn please help to review this PR, thanks

Reply:
If my suggestion is acceptable, please create another PR to fix ReplicaSet-Controller.